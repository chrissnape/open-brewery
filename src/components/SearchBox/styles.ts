import { StyleSheet } from 'react-native';
import { borderColour, fontSize, padding } from '../../utils/styles';

export default StyleSheet.create({
  icon: {
    fontSize: 20,
  },
  input: {
    backgroundColor: '#fff',
    borderColor: borderColour,
    borderWidth: 1,
    flex: 1,
    fontSize: fontSize + 2,
    marginRight: 10,
    padding: padding,
  },
  row: {
    alignItems: 'center',
    flexDirection: 'row',
  },
});
