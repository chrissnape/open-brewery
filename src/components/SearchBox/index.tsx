import React, { FC } from 'react';
import { TextInput, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import styles from './styles';

type Props = {
  queryString: string;
  onChange: (text: string) => void,
  onSubmit: () => void;
}

const SearchBoxComponent: FC <Props> = ({ queryString, onChange, onSubmit }): JSX.Element => (
  <View style={styles.row}>
    <TextInput
      style={styles.input}
      blurOnSubmit onChangeText={onChange}
      onSubmitEditing={onSubmit}
      placeholder="Search city..."
      value={queryString}
    />
    <Ionicons style={styles.icon} name="search-outline" />
  </View>
);

export default SearchBoxComponent;
