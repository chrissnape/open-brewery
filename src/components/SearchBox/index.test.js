import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from './index';

const props = {
  queryString: 'Boulder',
  onChange: () => {},
  onSubmit: () => {},
}

describe('<SearchBox />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SearchBox { ...props } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
