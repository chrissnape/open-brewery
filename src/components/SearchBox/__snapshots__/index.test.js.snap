// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Search city..."
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderWidth": 1,
        "flex": 1,
        "fontSize": 18,
        "marginRight": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value="Boulder"
  />
  <Text />
</View>
`;
